### DISCUSSION TITLE
Local Play without Internet

### DISCUSSION THREAD

**Discussion by RalphM77** [Thyraz/Sonos-Kids-Controller#10] [Q&A]
Dear all

Do you see a way the Kids Controller can be set up to play stuff without an internet connection e.g.over the headphone jack for example.
Would be great when travelling.

Best regards

Ralph

**Comment by NoTrace-Kai** [comment]
That would be a great function ;)

**Comment by Thyraz** [comment]
Hi Ralph,

This software is just a frontend for node-sonos-http-api.
It doesn't have any playback capabilities on it's own.

And node-sonos-http-api relies on the Sonos system and its library (and connected services).
So this won't work as long as you don't have a connection to your Sonos system.
And Sonos on the other hand can't play files from streaming services without an internet connection (only local library).

This means the whole software needs to be changed to control a local audioplayer and directly connected speakers instead of using your Sonos system.
And you need to provide all media as local files.

To be honest this is quite the opposite of what the software currently is about:
A simple controller for the Sonos system without the need to use a mobile phone, tablet or pc.
(And a box that can be used legally with your streaming services, without the need to rip the media files.)

I understand the intention, but if you want to keep the current state AND the possibility for offline music through a headjack, you're talking about managing 2 different media libraries for being at home and mobile use.

My simple solution when we're in the car: The kids have to accept, that they have to ask us to start some music from our mobile phones or tablets. ;)

It's not that hard to adjust the software to control a local media player that has some kind of http API instead of Sonos.
See this issue: https://github.com/Thyraz/Sonos-Kids-Controller/issues/7#event-4149563383

But as my use case is easy handling of audiobook streaming (and I'm a busy dad), I'm afraid this isn't something that I'll be building into the software.
Like I said, it also won't work well together with the current implementation as we're talking about 2 different library types (Sonos system vs. local file playback through speakers / headphones)

Regards,
Tobias
